app/views/comfy/admin/cms/fragments/_form_fragment_attachments.html.haml

```diff
@@ -4,9 +4,19 @@
       - variant = attachment.variant(Comfy::Cms::File::VARIANT_SIZE[:thumb])
       - thumb = image_tag(url_for(variant), size: "200x150")
     .fragment-attachment.btn-group.btn-group-sm.mb-1
-      - filename = attachment.filename.to_s
-      - truncated_filename = truncate(filename, length: 40, omission: "...#{filename.last(10)}")
-      = link_to truncated_filename, attachment, data: {toggle: "page-file-popover", content: thumb}, class: "btn btn-light text-truncate", target: "_blank"
+      :ruby
+        filename = attachment.filename.to_s
+        truncated_filename = truncate(filename, length: 40, omission: "...#{filename.last(10)}")
+        cms_page_file_link_tag = [
+          "{{ cms:page_file_link #{fragment_id}",
+          (", filename: \"#{filename}\"" if multiple),
+          (", as: image" if attachment.image?),
+          " }}".html_safe
+        ].compact.join
+      = link_to truncated_filename, attachment,
+                data: {toggle: "page-file-popover", content: thumb},
+                class: "btn btn-light text-truncate", target: "_blank",
+                ondragstart: "event.dataTransfer.setData('text/plain', '#{cms_page_file_link_tag}');"
```

Comment:
Don't think we can do inline JS like that if CSP is enabled. Let's move that into `data` attribute. Also how would one even know that this is something draggable?

Reply:
Done. I didn't write in CoffeeScript, but it's only a few line so perhaps you could rewrite it if you really want it to be CoffeeScript?

> Also how would one even know that this is something draggable?

True that it's not very discoverable, but it does have `cursor: pointer` on hover, and I've now made a thumbnail display for images while dragged.